Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -399,6 +401,9 @@ private void LogAccess(Entity<AccessReaderComponent> ent, EntityUid accessor)
             && idCard.Comp is { BypassLogging: false, FullName: not null })
             name = idCard.Comp.FullName;
 
+        if (TryComp<NameIdentifierComponent>(accessor, out var nameIdentifier))
+            name = nameIdentifier.FullIdentifier;
+
```

Comment:
remember i said you should put this check before the idCard one

Reply:
oh yes, makes sense